catroid/src/main/java/org/catrobat/catroid/stage/ShowTextActor.java

```diff
@@ -92,14 +91,9 @@ public static String convertToEnglishDigits(String value) {
 	public void draw(Batch batch, float parentAlpha) {
 		DataContainer dataContainer = ProjectManager.getInstance().getCurrentlyPlayingScene().getDataContainer();
 
-		List<UserVariable> projectVariableList = dataContainer.getProjectVariables();
-		Map<Sprite, List<UserVariable>> spriteVariableMap = dataContainer.getSpriteVariableMap();
-		List<UserVariable> spriteVariableList = spriteVariableMap.get(sprite);
-		List<UserVariable> userBrickVariableList = dataContainer.getOrCreateVariableListForUserBrick(userBrick);
-
-		drawVariables(projectVariableList, batch);
-		drawVariables(spriteVariableList, batch);
-		drawVariables(userBrickVariableList, batch);
+		drawVariables(dataContainer.getProjectUserVariables(), batch);
+		drawVariables(dataContainer.getSpriteUserVariables(sprite), batch);
+		drawVariables(dataContainer.getUserBrickUserVariables(userBrick), batch);
```

Comment:
`java.lang.NullPointerException: This map does not allow null keys`

Reply:
for now I went with option 2.